engine/src/main/java/org/hibernate/search/engine/impl/WorkPlan.java

```diff
@@ -111,12 +111,36 @@ public int size() {
 	 * @param typeIdentifier The entity class for which to retrieve the work
 	 *
 	 * @return returns (and creates if needed) the {@code PerClassWork} from the {@link #byClass} map.
+	 * @throws SearchException if the specified type is not @Indexed nor hosts any @ContainedIn
 	 */
 	private PerClassWork getClassWork(String tenantId, IndexedTypeIdentifier typeIdentifier) {
```

Comment:
Ah and one additional question: I suppose you kept this one because there are cases where it makes sense to use it?

It's not very obvious when we should use one or the other, maybe a comment would help understanding when it's necessary to use the one throwing an exception?

Reply:
Yeah, there are other places that need to access the class work, but have nothing to do with `@ContainedIn`, and thus do not care about/do not expect non-configured types.

I don't expect us to add other uses of this method (this code will be removed in 6), so I'm not really sure what to add to the javadoc... Essentially you should use the one that throws an exception when you want it to throw an exception if the entity type is not configured in Search, and the other when you want to handle the problem of non-configured entity types yourself...

Reply:
All good then.